confluence/server/scio_search/src/main/java/com/askscio/atlassian_plugins/confluence/impl/Utils.java

```diff
@@ -0,0 +1,112 @@
+package com.askscio.atlassian_plugins.confluence.impl;
+
+import static com.askscio.atlassian_plugins.confluence.impl.MyPluginComponentImpl.PLUGIN_STATUS_KEY;
+import static com.askscio.atlassian_plugins.confluence.impl.MyPluginComponentImpl.PLUGIN_STATUS_LAST_UPDATED_KEY;
+
+import com.atlassian.cache.Cache;
+import com.atlassian.cache.CacheLoader;
+import com.atlassian.extras.common.log.Logger;
+import com.atlassian.sal.api.pluginsettings.PluginSettings;
+import com.atlassian.sal.api.user.UserManager;
+import com.atlassian.sal.api.user.UserProfile;
+import com.google.gson.Gson;
+import java.io.Serializable;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+import org.codehaus.jackson.annotate.JsonAutoDetect;
+import org.codehaus.jackson.annotate.JsonAutoDetect.Visibility;
+
+public class Utils {
+
+  private static final Logger.Log logger = Logger.getInstance(Utils.class);
+  private static final Gson gson = new Gson();
+  private static final int PLUGIN_STATUS_UPDATE_INTERVAL_SECONDS = 60; // in seconds
+  private static final DateTimeFormatter formatter = DateTimeFormatter.ISO_LOCAL_DATE_TIME.withZone(
+      ZoneId.from(ZoneOffset.UTC));
+
+  public static boolean isCurrentUserAdmin(UserManager userManager) {
+    final UserProfile profile = userManager.getRemoteUser();
+    return profile != null && userManager.isSystemAdmin(profile.getUserKey());
+  }
+
+  public static void validateUserIsAdmin(UserManager userManager) {
+    if (!isCurrentUserAdmin(userManager)) {
+      throw new UnauthorizedException("Unauthorized");
+    }
+  }
+
+  public static PluginStatus getPluginStatus(PluginSettings pluginSettings) {
+    String pluginStatusStr = (String) pluginSettings.get(PLUGIN_STATUS_KEY);
+    if (pluginStatusStr == null) {
+      return new PluginStatus();
+    }
+    return gson.fromJson(pluginStatusStr, PluginStatus.class);
+  }
+
+  public static void updatePluginStatus(Cache<String, Optional<String>> pluginSettingsCache,
+      PluginSettings pluginSettings,
+      int responseCode) {
+    String pluginStatusStr = pluginSettingsCache.get(PLUGIN_STATUS_KEY).orElse(null);
+    PluginStatus pluginStatus = pluginStatusStr == null ? new PluginStatus() : gson.fromJson(
+        pluginStatusStr, PluginStatus.class);
+    String currentTimeStr = formatter.format(Instant.now());
```

Comment:
what is the timezone on this one? might be good to force UTC.

Reply:
Yes, we use UTC tz for the formatter